File: src/main/java/io/gittul/domain/user/dto/UserProfileResponse.java
package io.gittul.domain.user.dto;

import io.gittul.domain.user.entity.User;

public record UserProfileResponse(
        Long id,
        String nickname,
        String email,
        String profileImage
) {
    public static UserProfileResponse of(User user) {
        return new UserProfileResponse(
                user.getUserId(),
                user.getUserName(),
                user.getEmail(),
                user.getProfileImage()
        );
    }
}

File: app/src/main/kotlin/io/gittul/app/domain/user/dto/UserProfileResponse.kt
package io.gittul.app.domain.user.dto

import io.gittul.core.domain.user.entity.User

data class UserProfileResponse(
    val id: Long,
    val nickname: String,
    val email: String,
    val profileImage: String?
) {
    companion object {
        fun of(user: User): UserProfileResponse {
            return UserProfileResponse(
                user.userId,
                user.userName,
                user.email,
                user.profileImage
            )
        }
    }
}
